packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -174,7 +175,10 @@ class _RenderCompositionCallback extends RenderProxyBox {
 /// [TextInputFormatter]s instead for as-you-type text modification.
 ///
 /// If both the [text] and [selection] properties need to be changed, set the
-/// controller's [value] instead.
+/// controller's [value] instead. In some situations, only setting the controller's
+/// [text] property seems sufficient, but is not. For example, to clear
+/// a [TextField], it is insufficient to set [TextEditingController.text]
+/// to the empty string. Instead, use [TextEditingController.clear].
```

Comment:
Can you explain the "why" behind this? It would be good to explain what happens to the TextSelection when `.clear` is called versus when setting the `text` to empty string, and how that could cause a bug.